dask/dataframe/tests/test_groupby.py

```diff
@@ -1091,6 +1091,84 @@ def test_aggregate_dask():
                 assert len(other.dask) == len(result2.dask)
 
 
+# @pytest.mark.parametrize("axis", [0, 1])
+@pytest.mark.parametrize("method", ["ffill", "bfill"])
+@pytest.mark.parametrize("limit", [None, 1, 4])
+def test_fillna(method, limit):
+    df = pd.DataFrame(
+        {
+            "A": [1, 1, 2, 2],
+            "B": [np.nan, 3, np.nan, 4],
+            "C": [4, np.nan, 5, np.nan],
+            "D": [6, np.nan, 7, np.nan],
+        }
+    )
+    ddf = dd.from_pandas(df, npartitions=2)
+    assert_eq(
+        df.groupby("A").fillna(0),
+        ddf.groupby("A").fillna(0),
+    )
+    assert_eq(
+        df.groupby("A").B.fillna(0),
+        ddf.groupby("A").B.fillna(0),
+    )
+    assert_eq(
+        df.groupby("A").fillna(method=method, limit=limit),
+        ddf.groupby("A").fillna(method=method, limit=limit),
+    )
+
+    with pytest.raises(NotImplementedError):
+        ddf.groupby("A").fillna({"A": 0})
+
+    with pytest.raises(NotImplementedError):
+        ddf.groupby("A").fillna(pd.Series(dtype=int))
+
+    with pytest.raises(NotImplementedError):
+        ddf.groupby("A").fillna(pd.DataFrame)
```

Comment:
Now that I think of it, we might be able to get fillna with a dict/pandas object working with relatively little effort.

We could still defer on a Dask Series/DF, however.

Reply:
In our discussion yesterday, we noticed that value=dict creates a multi-indexed-DF. So, we decided to open a new issue to implement value = pandas dict/series/dataframe. I've added a TODO in the first comment, hence resolving this. :)